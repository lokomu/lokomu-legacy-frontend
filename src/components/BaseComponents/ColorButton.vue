<template>
  <button
    type="button"
    class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center"
    :class="[color, focusRingColor]"
  >
    {{ text }}
  </button>
</template>

<script setup lang="ts">
type ButtonColor = "red" | "green" | "blue";

const props = withDefaults(
  defineProps<{
    text: string;
    buttonColor?: ButtonColor;
  }>(),
  {
    buttonColor: "blue",
  },
);

const colors = {
  red: "from-pink-500 via-pink-600 to-pink-700",
  green: "from-green-400 via-green-600 to-green-700",
  blue: "from-floyd-400 via-floyd-600 to-floyd-700",
};
const ringColors = {
  red: "focus:ring-pink-300 dark:focus:ring-pink-800",
  green: "focus:ring-green-300 dark:focus:ring-green-800",
  blue: "focus:ring-floyd-300 dark:focus:ring-floyd-800",
};

const color = colors[props.buttonColor];
const focusRingColor = ringColors[props.buttonColor];
</script>
