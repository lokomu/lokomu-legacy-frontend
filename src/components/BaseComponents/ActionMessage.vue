<template>
  <div class="bg-white rounded-lg shadow dark:bg-gray-700 p-4">
    <h3 class="text-xl text-center font-semibold dark:text-white mb-4">
      {{ title }}
    </h3>
    <p class="text-base text-center text-gray-500 dark:text-gray-400 mb-4">
      {{ message }}
    </p>
    <div class="flex justify-center">
      <ColorButton
        :text="actionText"
        button-color="red"
        @click="emit('action')"
      />
      <ColorButton
        id="cancel-button"
        class="ml-8"
        :text="$t('button.action_button')"
        button-color="blue"
        @click="emit('close')"
      />
    </div>
  </div>
</template>

<script setup lang="ts">
import ColorButton from "@/components/BaseComponents/ColorButton.vue";

defineProps<{
  title: string;
  message: string;
  actionText: string;
}>();

const emit = defineEmits<{
  close: [];
  action: [];
}>();
</script>
