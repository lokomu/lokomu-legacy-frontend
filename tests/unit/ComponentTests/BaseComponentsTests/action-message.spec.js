import { mount } from '@vue/test-utils';
import ActionMessage from '@/components/BaseComponents/ActionMessage.vue';
import ColorButton from '@/components/BaseComponents/ColorButton.vue';
import {describe, it, expect} from "vitest";

describe('ActionMessage component', () => {
    const wrapper = mount(ActionMessage, {
        props: {
            title: 'Test Title',
            message: 'Test Message',
            actionText: 'Test Action',
        },
        global: {
            components: {
                ColorButton,
            },
        },
    });

    it('renders title, message, and action buttons correctly', () => {
        expect(wrapper.find('h3').text()).toBe('Test Title');
        expect(wrapper.find('p').text()).toBe('Test Message');
        expect(wrapper.findAllComponents(ColorButton)).toHaveLength(2);
        expect(wrapper.findComponent({ name: 'ColorButton', props: { text: 'Test Action' } })).toBeTruthy();
        expect(wrapper.find('#cancel-button')).toBeTruthy();
    });

    it('emits "close" event when close button is clicked', async () => {
        await wrapper.find('#cancel-button').trigger('click');
        expect(wrapper.emitted().close).toBeTruthy();
    });

    it('emits "action" event when action button is clicked', async () => {
        await wrapper.findComponent({ name: 'ColorButton', props: { text: 'Test Action' } }).trigger('click');
        expect(wrapper.emitted().action).toBeTruthy();
    });

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it("is instantiated", () => {
        expect(wrapper.exists()).toBeTruthy();
    });
});