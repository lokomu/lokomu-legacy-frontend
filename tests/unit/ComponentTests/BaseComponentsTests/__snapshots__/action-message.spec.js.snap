// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionMessage component > renders correctly 1`] = `
<div
  class="bg-white rounded-lg shadow dark:bg-gray-700 p-4"
>
  <h3
    class="text-xl text-center font-semibold dark:text-white mb-4"
  >
    Test Title
  </h3>
  <p
    class="text-base text-center text-gray-500 dark:text-gray-400 mb-4"
  >
    Test Message
  </p>
  <div
    class="flex justify-center"
  >
    <button
      class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center from-pink-500 via-pink-600 to-pink-700 focus:ring-pink-300 dark:focus:ring-pink-800"
      type="button"
    >
      Test Action
    </button>
    <button
      class="flex bg-gradient-to-br hover:bg-gradient-to-tl items-center text-white font-medium rounded-lg text-sm px-5 py-2.5 focus:ring-4 focus:outline-none text-center from-floyd-400 via-floyd-600 to-floyd-700 focus:ring-floyd-300 dark:focus:ring-floyd-800 ml-8"
      id="cancel-button"
      type="button"
    >
      button.action_button
    </button>
  </div>
</div>
`;
